docs: Convert from mkdocs to sphinx+myst_parser

gastmaier opened the pull request:
## Description

To standardize the look and features across the multiple analog-owned documentations, use sphinx+myst_parser with the harmonic theme.

Do minimal changes required by myst_parser, plus minor grammar review.
A lost feature is the glossary pop-up, instead, a glossary page is provided.
On the other hand, the pop-up from mkdocs don't work on pdfs and no glossary was provided.

I used toctree glob to auto glob pages, if ordering is important, please update the toctrees with each file name in the desired order.
I removed the _blank target from links, I will try to make it the default for external links through doctools.

user-guide/resources was moved to resources, since to match the hierarchical structure already defined.

Suggested deployment: https://analogdevicesinc.github.io/doctools/deploy.html
Version tags: https://analogdevicesinc.github.io/doctools/deploy.html#versioned

### Screenshots

Rendered as a pdf is provided here: 
[output.pdf](https://github.com/user-attachments/files/21098368/output.pdf)

```cd docs; adoc serve --builder=pdf --once```

And some screenshots
![image](https://github.com/user-attachments/assets/1c1f7de3-e3cc-4eb1-b56a-86041cd1db9f)
![image](https://github.com/user-attachments/assets/fb848e9b-bc10-4c71-810e-6262f2d5fa79)
![image](https://github.com/user-attachments/assets/33ff688e-2585-473f-89af-578cdaa5c00c)
```cd docs; adoc serve;```


## Type of Change

Documentation Stack

## Steps to test

```
cd docs
pip install -r requirements.txt
# build once
make html
# or (live server)
adoc serve
```

## Admonition

I changed the syntax of the admonitions to use myst style.
Admonition are not part of the markdown syntax, but a [Python Markdown](https://python-markdown.github.io/extensions/admonition/) extension and the syntax is discouraged by [PyMdown](https://facelessuser.github.io/pymdown-extensions/extensions/blocks/plugins/admonition/) (as is in main, this doc favors pymdownx)
reverting back to mkdocs from this pr does not require syntax change if [mystmd](https://mystmd.org/guide/admonitions) is used.

## Checklist

<!-- Mandatory -->
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings (e.g., in the developer tools console).
- [ ] I have run pre-commit checks (e.g., lint).
- [ ] I have assigned the PR to the appropriate person.
- [ ] I have divided the code changes into a reasonable number of commits with clear commit messages.

gastmaier committed:
docs: Convert from mkdocs to sphinx+myst_parser

To standardize the look and features across the multiple analog-owned
documentations, use sphinx+myst_parser with the harmonic theme.

Do minimal changes required by myst_parser, plus minor grammar review.
A lost feature is the glossary pop-up, instead, a glossary page is
provided.

Signed-off-by: Jorge Marques <jorge.marques@analog.com>

carolinemaymaguirecmm commented:
Hey @gastmaier 😀
We are not migrating CodeFusion Studio (CFS) documentation to reStructuredText at this time. Please feel free to reach out to us directly if you want to discuss it further.

gastmaier committed:
docs: Don't limit root toctree depth, small fix

Setting the maxdepth in the root toctree affects the depth of the pdf's
table of contents.

Signed-off-by: Jorge Marques <jorge.marques@analog.com>
